Create checkpoints for Day 2

Saunved opened the pull request:
Also some minor code fixes (non-interfering)

Saunved committed:
Use the tweet ID as the key

+ React best practice

Saunved committed:
Use user ID instead of idx

Saunved committed:
Remove the older ninth patch

Saunved committed:
Create patch for ninth checkpoint

nrkin commented on checkpoints/eight.patch:
```diff
@@ -0,0 +1,67 @@
+-            <Tweet tweet={tweet} /> <hr />
+-          </div>
++          // Task 5:
++          // We want to "map" each tweet to the Tweet UI component here.
```
We want to have UI corresponding to each tweet - an array of components corresponding with the array of tweets. Which JS Array function does this?

vishwagosalia committed:
Sixth Checkpoint

msomya committed:
Create seventh checkpoint patch

This checkpoint walks you through the application of how database
interacts with the code.

hritik5102 reviewed (approved):
LGTM

rishabhj1717 committed:
Added checkpoint 6

rishabhj1717 committed:
Added checkpoint 7

rishabhj1717 committed:
Add savepoint 8

rishabhj1717 committed:
Add savepoint 9

a-pavaskar committed:
Add patch file for the 10th checkpoint

This was cherry picked from `feature/checkpoints`

rishabhj1717 committed:
Add Savepoint for backend 10

rishabhj1717 committed:
Update savepoint for checkpoint 9

Saunved committed:
Create checkpoint 10 FE patch

rishabhj1717 committed:
Add FE savepoint for checkpoint 10
